### DISCUSSION TITLE
script  DeployFundMe.s.sol.....  no such file or directory (os error 2)

### DISCUSSION THREAD

**Discussion by AnaAse** [Cyfrin/foundry-full-course-cu#3137] [General]
After reaching here and chating with Chat GPT in a conversation that seems to have come to a loop. The following prompt is returning an error:

forge script script/DeployFundMe.s.sol:DeployFundMe --rpc-url http://127.0.0.1:8545 --account defaultKey --sender "adress provided from encripting" --broadcast -vvvv

Im not sure if I missed at some point storing/creating a DeployFundMe.s.sol file. I cannot recall it. I would appreciate some light to this.
And at this point in time, Happy New Year 2025 in Smart Contract learning

**Comment by EngrPips** [comment]
What error are you getting @AnaAse

**Comment by KMean** [comment]
@AnaAse Of course you must have the DeployFundMe.s.sol to call it. 
As a side note you can run scripts without providing the --sender flag if the --account flag is specified. Here's an example:
```bash
forge script script/DeployFundMe.s.sol:DeployFundMe --rpc-url http://127.0.0.1:8545/ --account defaultKey --broadcast
```
For this to work, ensure that you've already imported an account with a mock private key into your keystore. For example:
```bash
cast wallet import defaultKey --interactive
Enter private key: <0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80>
Enter password: <YOUR-PASSWORD>
`defaultKey` keystore was saved successfully. Address: 0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266
```
This allows forge to use the defaultKey account as the sender automatically.

**Reply by EngrPips** [reply]
great heads-up.

**Reply by 0xPeregrin** [reply]
I am given the same error too so I created the file DeployFundMe.s.sol in folder script to call it  but still given the same error again.

I think Mr.Patrick has not covered FundMe part before moving on with this video or not necessary? I am not sure.

Also, do I need to deploy FundMe like SimpleStorage, I am kinda lost.

Thank you! @KMean

**Reply by KMean** [reply]
> I am given the same error too so I created the file DeployFundMe.s.sol in folder script to call it  but still given the same error again.
> 
> 
> 
> I think Mr.Patrick has not covered FundMe part before moving on with this video or not necessary? I am not sure.
> 
> 
> 
> Also, do I need to deploy FundMe like SimpleStorage, I am kinda lost.
> 
> 
> 
> Thank you! @KMean 

If the error is ```no such file or directory (os error 2)``` make absolutely sure that you are running your commands from the correct (root) folder of your project. If that still gives you an error post a snapshot of your folder structure and terminal to see from where you're running commands and where your files are. For the deployment question it depends in what part of the FundMe lesson you are. [At the beginning](https://updraft.cyfrin.io/courses/foundry/foundry-fund-me/writing-tests-for-solidity-smart-contracts) Patrick just deploys the contract from the test. [After](https://updraft.cyfrin.io/courses/foundry/foundry-fund-me/advanced-deploy-scripts) it teaches you how to deploy it via scripts.

**Reply by EngrPips** [reply]
Phenomenal heads-up.
